stacks/styleguide.prx.org.yml

```diff
@@ -0,0 +1,208 @@
+  EnvironmentType:
+    Type: String
+  EnvironmentTypeAbbreviation:
+    Type: String
+  EcrRegion:
+    Type: String
+  SecretsBase:
+    Type: String
+  ContainerMemory:
+    Type: String
+  ContainerCpu:
+    Type: String
+  # Shared ENV #################################################################
+  # Styleguide ECS ################################################################
+  StyleguideEcrImageTag:
+    Type: String
+  # Styleguide ENV ################################################################
+  StyleguideSecretsVersion:
+    Type: String
+Resources:
+  StyleguideLogGroup:
+    Type: "AWS::Logs::LogGroup"
+    Properties:
+      RetentionInDays: 14
+  StyleguideALBTargetGroup:
+    Type: "AWS::ElasticLoadBalancingV2::TargetGroup"
+    Properties:
+      HealthCheckIntervalSeconds: 15
+      HealthCheckTimeoutSeconds: 5
+      HealthyThresholdCount: 3
+      UnhealthyThresholdCount: 3
+      HealthCheckPath: /
+      Name: !Sub styleguide-${EnvironmentTypeAbbreviation}-${VPC}
+      Port: 80
+      Protocol: HTTP
+      TargetGroupAttributes:
+        - Key: deregistration_delay.timeout_seconds
+          Value: "15"
+      Tags:
+        - Key: Project
+          Value: !FindInMap [Shared, Project, name]
+        - Key: Environment
+          Value: !Ref EnvironmentType
+        - Key: Name
+          Value: !Sub Styleguide-${EnvironmentType}
+        - Key: "prx:cloudformation:stack-name"
+          Value: !Ref AWS::StackName
+        - Key: "prx:cloudformation:stack-id"
+          Value: !Ref AWS::StackId
+      VpcId: !Ref VPC
+  # ALB Listener Rules
+  StyleguideALBHTTPSHostWildcardListenerRule:
+    Type: "AWS::ElasticLoadBalancingV2::ListenerRule"
+    Properties:
+      Actions:
+        - TargetGroupArn: !Ref StyleguideALBTargetGroup
+          Type: forward
+      Conditions:
+        - Field: host-header
+          Values:
+            - styleguide.*
+      ListenerArn: !Ref PlatformALBHTTPSListenerArn
+      Priority: !Join ["", [!Ref PlatformALBListenerPriorityPrefix, "00"]]
+  StyleguideALBHTTPHostWildcardListenerRule:
+    Type: "AWS::ElasticLoadBalancingV2::ListenerRule"
+    Properties:
+      Actions:
+        - TargetGroupArn: !Ref StyleguideALBTargetGroup
+          Type: forward
+      Conditions:
+        - Field: host-header
+          Values:
+            - styleguide.*
+      ListenerArn: !Ref PlatformALBHTTPListenerArn
+      Priority: !Join ["", [!Ref PlatformALBListenerPriorityPrefix, "00"]]
+  # ECS Service
+  StyleguideTaskDefinition:
+    Type: "AWS::ECS::TaskDefinition"
+    Properties:
+      ContainerDefinitions:
+        - Cpu: !Ref ContainerCpu
+          Environment:
+            - Name: APP_NAME
+              Value: "styleguide"
+            - Name: APP_ENV
+              Value: !Ref EnvironmentTypeAbbreviation
+            - Name: AWS_SECRETS_BASE
+              Value: !Ref SecretsBase
+            - Name: AWS_SECRETS_VERSION
+              Value: !Ref StyleguideSecretsVersion
```

Comment:
I'm fuzzy on how the secrets are set up, is this part of the process of creating the initial configuration in S3 described here? https://github.com/PRX/Infrastructure/tree/master/cd#staging-deploy

Reply:
That's the S3 object version set automagically by CD - coming out of the `s3://prx-infrastructure-us-east-1-config/template-config-staging.zip`.  And there's some sort of lambda trigger on the S3 bucket that will update the zip whenever you change the secrets file in S3.